compiler/rustc_next_trait_solver/src/canonicalizer.rs

```diff
@@ -44,8 +45,15 @@ pub struct Canonicalizer<'a, D: SolverDelegate<Interner = I>, I: Interner> {
     canonicalize_mode: CanonicalizeMode,
 
     variables: &'a mut Vec<I::GenericArg>,
+    variable_lookup_table: HashMap<I::GenericArg, usize>,
+
     primitive_var_infos: Vec<CanonicalVarInfo<I>>,
     binder_index: ty::DebruijnIndex,
```

Comment:
Can you group these by the `// immutable` / `// mutable` above? I think that grouping was a great idea.

Also, could leave a comment explaining why `variables` is not mutable b/c it represents no *new* computational state.